src/cpp/proxy/engine.cpp

```diff
@@ -450,6 +451,7 @@ class Engine::Private : public QObject
 
 		// proxysession will take it from here
 		// TODO: use callbacks for performance
+		reqSessionConnectionMap.erase(rs);
 		rs->disconnect(this);
```

Comment:
If there are no more qt connections between engine and RequestSession then this line can be removed.